src/core.jsx

```diff
@@ -31,21 +32,17 @@ export default config => WrappedComponent => class extends Component {
   }
 
   format = (nextProps) => {
-    if (!this.context.store
-      || !this.context.store.getState()
-      || !this.context.store.getState().locale) {
-      throw new Error('/ HOC k-intl / locale is not readable. make sure that this one is available at the root of your redux store')
-    }
-
-    /* not change labels when the sub store local not change */
-    if (this.context.store.getState().locale === this.locale) return
-    this.locale = this.context.store.getState().locale
-
+    /* take locale on `config.locale` reducer */
+    const locale = getLocale(this.context)
+    const labels = format(locale)(
+      config,
+      nextProps || this.props,
+    )
+    /* not change labels when the formated labels is identical */
+    if (shallowEqual(this.state.injectedProps.labels, this.labels)) return
```

Comment:
- you want to compare injectedProps.labels and labels (not this.labels)
 - you can remove this.labels